bnd.bnd

```diff
@@ -0,0 +1,10 @@
+Bundle-SymbolicName: org.jspecify.jspecify
+Bundle-Name: JSpecify annotations
+Bundle-Vendor: The JSpecify Authors
+Bundle-Description: An artifact of well-specified annotations to power static analysis checks and JVM language interop.
+Bundle-DocURL: https://jspecify.dev/docs/start-here
+Bundle-License: https://www.apache.org/licenses/LICENSE-2.0
+-exportcontents: org.jspecify.*
+-reproducible: true
+-noextraheaders: true
+-snapshot: SNAPSHOT
```

Comment:
Is this line something that we'd need to change when we publish a proper 1.0.0 release? Or is it used only when the version number is already a snapshot?

(Even if we need to change it for the release, I am happy to have it set in the main branch.)

Reply:
Also, extremely minor: I see that this file has to trailing newline. (Vim keeps re-adding one when I mess with it :)) I assume that we can include one without causing any problems?

Reply:
It looks like, with or without this line, we get:

```
Bundle-Version: 0.0.0.SNAPSHOT
Export-Package: org.jspecify.annotations;version="0.0.0.SNAPSHOT"
```

And, with the version in `gradle.properties` bumped to `1.0.0`, we likewise get the same thing with or without this line:

```
Bundle-Version: 1.0.0
Export-Package: org.jspecify.annotations;version="1.0.0"
```

So maybe we don't need it, but maybe it's not causing any harm, either?

Reply:
> Is this line something that we'd need to change when we publish a proper 1.0.0 release? Or is it used only when the version number is already a snapshot?

I think it can stay as is ([see meaning of -snapshot instruction](https://bnd.bndtools.org/instructions/snapshot.html))

The `Bundle-Version` for the OSGi Manifest.mf is taken from `gradle.properties` `version` attribute, which is currently set to version = 0.0.0-SNAPSHOT:

https://github.com/jspecify/jspecify/blob/f0af7de6f72d01e3a8f3ffc9f528540de770f987/gradle.properties#L2

Given the current `bnd.bnd` this results in:

`jspecify-0.0.0-SNAPSHOT.jar`
and
`Bundle-Version: 0.0.0.SNAPSHOT`

Locally I simulated a 1.0.0 release for testing I changed gradle.properties version to:
`version = 1.0.0`

This resulted in:

`jspecify-1.0.0.jar`
and
`Bundle-Version: 1.0.0`

(so SNAPSHOT was removed and proper version was used)

In conclusion: I think this is what you would do for a release too, right?

Reply:
Thanks, I should have been looking in the bnd docs instead of the OSGi docs... :)

Given that "The default is no substitution," it sounds like this line is a no-op. But that would also mean that it's harmless. Since it's easier to leave it how it is, I plan to do that.

Reply:
> Thanks, I should have been looking in the bnd docs instead of the OSGi docs... :)

No problem. It can be confusing. Thanks for all the pointers.
I will be around for another 2 hours some more changes are required.